packages/app/src/app/pages/Dashboard/Content/utils.ts

```diff
@@ -39,3 +39,38 @@ export function getPossibleTemplates(
     template => template.id
   );
 }
+
+function copyArray(source) {
+  let index = -1;
+  const length = source.length;
+
+  const array = new Array(length);
+  while (++index < length) {
+    array[index] = source[index];
+  }
+  return array;
+}
+
+export function shuffleSeed(array, inputSeed) {
+  let seed = inputSeed || 1;
+
+  const random = () => {
+    const x = Math.sin(seed++) * 10000;
+    return x - Math.floor(x);
+  };
+
+  const length = array == null ? 0 : array.length;
+  if (!length) {
+    return [];
+  }
+  let index = -1;
+  const lastIndex = length - 1;
+  const result = copyArray(array);
```

Comment:
Looks like you want a shallow copy (right?), you can also do it like this:

```suggestion
  const result = [ ...array ]
```

Reply:
Do you know if `[...array] works with Proxy object?

Reply:
yep, it should work